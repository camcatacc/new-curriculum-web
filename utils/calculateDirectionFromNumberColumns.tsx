// TODO: Simplify this function
function calculateDirectionFromNumberColumns(i: number, total: number, numPerRow: number): "fromLeft" | "fromRight" | "fromBottom" {
	// If everything does not fit in a row
	if (numPerRow < total) {
		// We calculate the number of items in the last row
		let nItemsLastRow = total % numPerRow;
		// We calculate the position from which the items will be in the last row
		let threshold = total - nItemsLastRow;
		// And we calculate the item position in that row
		let posInNewRow = i - numPerRow;

		// If the threshold is smaller than the item position and there are remaining items to be placed enter this conditional
		// If not, then the default method can be applied
		if (threshold < i && nItemsLastRow !== 0) {
			// If there is only one item in the last row, it must come from the bottom
			if (nItemsLastRow === 1) return "fromBottom";
			// We use the default method BUT with the position of the item IN THE LAST ROW
			switch (posInNewRow % nItemsLastRow) {
				case 0:
					return "fromRight";
				case 1:
					return "fromLeft";
				default:
					return "fromBottom";
			}
		}
	}

	switch (
		i % numPerRow //
	) {
		case 0:
			return "fromRight";
		case 1:
			return "fromLeft";
		default:
			return "fromBottom";
	}
}

export default calculateDirectionFromNumberColumns;
