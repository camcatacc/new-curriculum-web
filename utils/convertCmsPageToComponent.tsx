// Modules
import React from "react";

// Elements
import Abilities from "components/templates/Abilities/Abilities";
import Presentation from "components/templates/Presentation/Presentation";
import FindMe from "components/templates/FindMe/FindMe";

// Definitions
import type { ContentfulEntry } from "interfaces/cms/contentful";
import type { CmsPage } from "interfaces/cms/pages/Page";
import type { CmsPresentationPage } from "interfaces/cms/pages/Presentation";
import type { CmsAbilitiesPage } from "interfaces/cms/pages/Abilities";
import type { CmsFindMePage } from "interfaces/cms/pages/FindMe";

// Functions
const convertCmsPageToComponent = (page: ContentfulEntry<CmsPage>) => {
	switch (page.fields.type) {
		case "PresentationPage":
			return presentationCmsToComponent(page.fields.content.fields);
		case "AbilitiesPage":
			return abilitiesCmsToComponent(page.fields.content.fields);
		case "FindMePage":
			return findMeCmsToComponent(page.fields.content.fields);
	}
};

const presentationCmsToComponent = (page: CmsPresentationPage) => <Presentation {...page} />;
const abilitiesCmsToComponent = (page: CmsAbilitiesPage) => {
	const lists = page.abilitiesLists.map((abList) => {
		return {
			...abList.fields
		};
	});

	return <Abilities title={page.title} abilitiesLists={lists} />;
};

const findMeCmsToComponent = (page: CmsFindMePage) => <FindMe {...page} />;

export default convertCmsPageToComponent;
