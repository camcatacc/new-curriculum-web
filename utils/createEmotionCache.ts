// Modules
import createCache from "@emotion/cache";

// Functions
export default function createEmotionCache() {
	return createCache({ key: "css" });
}
