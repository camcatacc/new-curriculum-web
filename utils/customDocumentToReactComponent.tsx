// Modules
import React from "react";
import { BLOCKS, MARKS } from "@contentful/rich-text-types";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";

// Styles
import { secondary } from "styles/customColors";

// Elements
import { Paragraph } from "@hardru/curriculum-web-components";

// Definitions
import type { Document } from "@contentful/rich-text-types";
import type { Options } from "@contentful/rich-text-react-renderer";
import type { PropTypes, Theme, TypographyVariant } from "@material-ui/core";

export interface CustomDocumentToReactComponentOptions {
	theme: Theme;
	boldColor?: boolean;
	marginBottom?: 1 | 2 | 3 | 4 | 6 | 8 | 10;
	align?: PropTypes.Alignment;
	variant?: TypographyVariant;
}

// Function
function customDocumentToReactComponent(document: Document, { boldColor, align, variant, theme }: CustomDocumentToReactComponentOptions) {
	let options: Options = {
		renderNode: {
			[BLOCKS.PARAGRAPH]: (_: any, children: any) => (
				<div>
					<Paragraph variant={variant} align={align}>
						{children}
					</Paragraph>
				</div>
			)
		}
	};

	if (boldColor) {
		options = { ...options, renderMark: {} };
		options.renderMark![MARKS.BOLD] = (text) => <b style={{ color: theme.palette.secondary[theme.palette.type] }}>{text}</b>;
	}

	return documentToReactComponents(document, options);
}

export default customDocumentToReactComponent;
