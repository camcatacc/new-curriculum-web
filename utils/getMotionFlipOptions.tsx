// Definitions
import type { MotionProps, Variants } from "framer-motion";

export interface GetMotionFlipOptionsOptions {
	amount?: number;
	delay?: number;
}

// Functions
function getMotionFlipOptions(animated: boolean, { amount, delay }: GetMotionFlipOptionsOptions): MotionProps {
	if (!animated) return {};
	const variants = getMotionFlipVariants(animated, delay);
	return {
		variants: variants,
		viewport: {
			amount,
			once: true
		},
		initial: "offscreen",
		whileInView: "onscreen"
	};
}

export function getMotionFlipVariants(animated: boolean, delay?: number): Variants {
	if (!animated) return {};
	return {
		offscreen: {
			rotateY: 90,
			opacity: 0
		},
		onscreen: { rotateY: 0, opacity: 1, transition: { duration: 0.5, delay: delay } }
	};
}

export default getMotionFlipOptions;
