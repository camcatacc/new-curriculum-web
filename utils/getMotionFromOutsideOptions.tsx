// Definitions
import type { MotionProps, Variants } from "framer-motion";

export type AnimatedDirection = "fromLeft" | "fromRight" | "fromBottom";

// Functions
export function getOnScrollMotionFromOutsideOptions(animated: boolean, animatedDirection: AnimatedDirection, delay?: number): MotionProps {
	if (!animated) return {};

	const { initial, animation } = getMotionFromOutsideVariants(animatedDirection, delay);

	const variants: Variants = {
		offscreen: initial,
		onscreen: animation
	};

	return {
		variants,
		initial: "offscreen",
		whileInView: "onscreen",
		viewport: { once: true, amount: 0.6 }
	};
}

export function getMotionFromOutsideVariants(animatedDirection: AnimatedDirection, delay?: number): Variants {
	let initialX = 0,
		initialY = 0;

	switch (animatedDirection) {
		case "fromLeft":
			initialX = -50;
			break;
		case "fromRight":
			initialX = 50;
			break;
		case "fromBottom":
			initialY = 50;
			break;
	}
	return {
		initial: { x: initialX, y: initialY, opacity: 0 },
		animation: {
			x: 0,
			y: 0,
			opacity: 1,
			transition: {
				type: "tween",
				duration: 1,
				delay: delay,
				when: "beforeChildren"
			}
		}
	};
}

export default getOnScrollMotionFromOutsideOptions;
