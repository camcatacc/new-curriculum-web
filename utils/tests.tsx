// Modules
import React, { ReactElement } from "react";
import { ThemeProvider } from "@material-ui/styles";

// Testing modules
import { render, RenderOptions, RenderResult } from "@testing-library/react";

// Redux Modules
import { configureStore } from "@reduxjs/toolkit";

// Redux Components
import rootReducer from "redux/rootReducer";
import { Provider } from "react-redux";
import { defaultTestRootState } from "redux/store.test";

// Auxiliary functions
import { getMaterialTheme } from "styles/materialUi";

// Definitions
import type { RootState } from "redux/types";

// Functions
export const customRender = (ui: ReactElement, state: Partial<RootState> = defaultTestRootState, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		const store = configureStore({
			reducer: rootReducer,
			preloadedState: state
		});

		return (
			<ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>
				<Provider store={store}>{children}</Provider>
			</ThemeProvider>
		);
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
