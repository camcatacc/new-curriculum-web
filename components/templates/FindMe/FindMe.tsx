// Modules
import React from "react";
import customDocumentToReactComponent from "utils/customDocumentToReactComponent";

// Elements
import SectionTitle from "components/molecules/SectionTitle/SectionTitle";
import { Button } from "@material-ui/core";

// Definitions
import type { Document } from "@contentful/rich-text-types";

export interface FindMeProps {
	text: Document;
	title: string;
}

// Element
const FindMe = ({ text, title }: FindMeProps) => {
	const formattedText = customDocumentToReactComponent(text, { boldColor: true, align: "center" });

	return (
		<div className="flex flex-col justify-center">
			<SectionTitle>{title}</SectionTitle>
			{formattedText}
			<Button color="primary" variant="contained" href="mailTo:camachocateura1997@gmail.com" style={{ marginLeft: "auto", marginRight: "auto", marginTop: "2rem" }}>
				CONTACT ME
			</Button>
		</div>
	);
};

export default FindMe;
