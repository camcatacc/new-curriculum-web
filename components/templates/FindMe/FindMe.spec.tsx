// Modules
import React from "react";
import { BLOCKS, INLINES } from "@contentful/rich-text-types";

// Test Modules
import { customRender } from "utils/tests";

// Accessibility Tests
import { axe, toHaveNoViolations } from "jest-axe";

// Components
import FindMe, { FindMeProps } from "components/templates/FindMe/FindMe";
import { fireEvent } from "@testing-library/dom";

// APIs to be mocked
import emailService from "services/api/email";
import { act } from "@testing-library/react";

// Accessibility extension
expect.extend(toHaveNoViolations);

// Mocks
jest.mock("services/api/email");

const mockedEmailService = emailService as jest.Mock;

// Default Components
const defaultProps: FindMeProps = {
	title: "Find Me",
	text: {
		nodeType: BLOCKS.DOCUMENT,
		data: {},
		content: [
			{
				nodeType: BLOCKS.PARAGRAPH,
				content: [
					{
						nodeType: "text",
						value: "Do you want to work on a project with me or have any kind of doubt?",
						marks: [],
						data: {}
					}
				],
				data: {}
			},
			{
				nodeType: BLOCKS.PARAGRAPH,
				content: [
					{
						nodeType: "text",
						value: "You can find me in ",
						marks: [],
						data: {}
					},
					{
						nodeType: INLINES.HYPERLINK,
						content: [
							{
								nodeType: "text",
								value: "contact@gmail.com",
								marks: [],
								data: {}
							}
						],
						data: {
							uri: "mailto:contact@gmail.com"
						}
					},
					{
						nodeType: "text",
						value: ".",
						marks: [],
						data: {}
					}
				],
				data: {}
			}
		]
	}
};

// Custom rendering
function renderComponent(newProps?: object) {
	const props = { ...defaultProps, ...newProps };
	return customRender(<FindMe {...props} />);
}

describe("FindMe", () => {
	beforeEach(() => {
		mockedEmailService.mockReset().mockResolvedValue({ status: 200 });
	});
	it("Snapshot and accessibility tests", async () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});

	it("On Button click, the modal is opened and its content is rendered well", async () => {
		const { container, getByRole } = renderComponent();

		const button = getByRole("button");
		fireEvent.click(button);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});

	describe("On Form Change...", () => {
		it("On Form Change, values change", () => {
			const { container, getByRole, getByLabelText } = renderComponent();

			const modalButton = getByRole("button");

			fireEvent.click(modalButton);

			const nameInput = getByLabelText("Name") as HTMLInputElement;
			fireEvent.change(nameInput, { target: { value: "David" } });

			expect(nameInput.value).toBe("David");
			expect(container).toMatchSnapshot();
		});

		it("On Form Change, if cancel button is pressed, values are not saved", () => {
			const { container, getByRole, getByLabelText, getByText } = renderComponent();

			const modalButton = getByRole("button");

			fireEvent.click(modalButton);

			const nameInput = getByLabelText("Name") as HTMLInputElement;
			fireEvent.change(nameInput, { target: { value: "David" } });

			const cancelButton = getByText("Cancel");
			fireEvent.click(cancelButton);
			fireEvent.click(modalButton);

			expect(nameInput.value).toBe("");
			expect(container).toMatchSnapshot();
		});

		it("On Form Change, if SAVE button is pressed, values are saved", () => {
			const { container, getByRole, getByLabelText, getByText } = renderComponent();

			const modalButton = getByRole("button");

			fireEvent.click(modalButton);

			const nameInput = getByLabelText("Name") as HTMLInputElement;
			fireEvent.change(nameInput, { target: { value: "David" } });

			const saveButton = getByText("Finish it later");
			fireEvent.click(saveButton);
			fireEvent.click(modalButton);

			expect(nameInput.value).toBe("David");
			expect(container).toMatchSnapshot();
		});

		it("On Form Change, if SEND WITH ERROR button is pressed, an error appears", async () => {
			const { container, getByRole, findByText } = renderComponent();

			const modalButton = getByRole("button");

			fireEvent.click(modalButton);

			const errorButton = await findByText("Send with error");
			fireEvent.click(errorButton);

			const errorText = await findByText("This request fails after one second");

			expect(errorText).toBeInTheDocument();
			expect(container).toMatchSnapshot();
		});

		it("On Form Change, if SEND button is pressed and the request succeeds, the modal is closed", async () => {
			const { container, findByRole, findByText } = renderComponent();

			const modalButton = await findByRole("button");
			fireEvent.click(modalButton);

			const sendButton = await findByText("Send");
			fireEvent.click(sendButton);

			await act(async () => {
				await new Promise((resolve) => setTimeout(resolve, 500));
			});

			expect(sendButton).not.toBeInTheDocument();
			expect(container).toMatchSnapshot();
		});
	});
});
