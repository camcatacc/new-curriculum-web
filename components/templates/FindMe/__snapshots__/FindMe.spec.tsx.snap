// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindMe On Button click, the modal is opened and its content is rendered well 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe On Form Change... On Form Change, if SAVE button is pressed, values are saved 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe On Form Change... On Form Change, if SEND WITH ERROR button is pressed, an error appears 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe On Form Change... On Form Change, if SEND button is pressed and the request succeeds, the modal is closed 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe On Form Change... On Form Change, if cancel button is pressed, values are not saved 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe On Form Change... On Form Change, values change 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FindMe Snapshot and accessibility tests 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Find Me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          Do you want to work on a project with me or have any kind of doubt?
        </p>
      </div>
      <div>
        <p
          class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignCenter"
        >
          You can find me in 
          <a
            href="mailto:contact@gmail.com"
          >
            contact@gmail.com
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="flex mt-8 justify-center"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            CONTACT ME
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
